libcontainer/cgroups/fs/fs.go

```diff
@@ -133,46 +131,19 @@ func getCgroupRoot() (string, error) {
 		return cgroupRoot, nil
 	}
 
-	// slow path: parse mountinfo, find the first mount where fs=cgroup
-	// (e.g. "/sys/fs/cgroup/memory"), use its parent.
-	f, err := os.Open("/proc/self/mountinfo")
+	// slow path: parse mountinfo
+	mi, err := cgroups.ReadCgroupMountinfo()
 	if err != nil {
 		return "", err
 	}
-	defer f.Close()
-
-	var root string
-	scanner := bufio.NewScanner(f)
-	for scanner.Scan() {
-		text := scanner.Text()
-		fields := strings.Split(text, " ")
-		// Safe as mountinfo encodes mountpoints with spaces as \040.
-		index := strings.Index(text, " - ")
-		postSeparatorFields := strings.Fields(text[index+3:])
-		numPostFields := len(postSeparatorFields)
-
-		// This is an error as we can't detect if the mount is for "cgroup"
-		if numPostFields == 0 {
-			return "", fmt.Errorf("mountinfo: found no fields post '-' in %q", text)
-		}
-
-		if postSeparatorFields[0] == "cgroup" {
-			// Check that the mount is properly formatted.
-			if numPostFields < 3 {
-				return "", fmt.Errorf("Error found less than 3 fields post '-' in %q", text)
-			}
-
-			root = filepath.Dir(fields[4])
-			break
-		}
-	}
-	if err := scanner.Err(); err != nil {
-		return "", err
-	}
-	if root == "" {
+	if len(mi) < 1 {
 		return "", errors.New("no cgroup mount found in mountinfo")
 	}
 
+	// Get the first cgroup mount (e.g. "/sys/fs/cgroup/memory"),
+	// use its parent directory.
+	root := filepath.Dir(mi[0].Root)
```

Comment:
Should we be using `path` instead of `filepath`? I think everything here assumes Linux separators, in which case `path` would be a better fit.

Reply:
This question can be applied to lots of code :)

I'd rather be careful and use `filepath.Dir`. I understand `path.Dir` should be somewhat faster, but we're not calling it 10.000 times here.

I think it's a separate issue and can be addressed globally (replace all `filepath.XXX` with `path.XXX` or so).

Reply:
Ahh, cpuset.go already uses both `path.Dir` and `filepath.Dir`. Surely this can be unified; let me see...

Reply:
Yes, makes sense to do separate. My eye fell on it here, so sorry for the stray comment ☺️

Reply:
Yeah we generally use `filepath` for our own sanity, but we can do a switch in a separate PR if there is a strong reason to do so.